projects/plugins/jetpack/extensions/blocks/payments-intro/edit.js

```diff
@@ -12,20 +12,37 @@ import defaultVariations from './variations';
 export default function JetpackPaymentsIntroEdit( { name, clientId, className } ) {
 	const patternFilter = pattern => pattern.categories?.includes( 'shop' );
 
-	const { blockType, hasInnerBlocks, hasPatterns } = useSelect( select => {
-		const { getBlocks, __experimentalGetAllowedPatterns } = select( blockEditorStore );
+	const { blockType, hasInnerBlocks, hasPatterns, settings } = useSelect( select => {
+		const { getBlocks, __experimentalGetAllowedPatterns, getSettings } = select( blockEditorStore );
 
 		return {
+			settings: getSettings(),
 			blockType: getBlockType( name ),
 			hasInnerBlocks: getBlocks( clientId )?.length > 0,
 			hasPatterns: __experimentalGetAllowedPatterns?.().filter( patternFilter ).length > 0,
 		};
 	} );
 
-	const { replaceBlock, selectBlock } = useDispatch( blockEditorStore );
+	const { replaceBlock, selectBlock, updateSettings } = useDispatch( blockEditorStore );
+
+	const maybeUpdateBlockVisibility = useCallback(
```

Comment:
maybe the name should imply that the block gets visible? Like `maybeSetBlockVisible`?

Reply:
Fixed in [5064fed](https://github.com/Automattic/jetpack/pull/30095/commits/5064fed9981dedcd09be49821486a13bd815788c)